### PR TITLE
Adding mi docker test changes

### PR DESCRIPTION
**Pull request by RidmiR**
## Purpose
Purpose of this PR is to run samples tests inside an MI Docker container. 

## Approach
These tests are running on Testgrid Devenviornment (https://testgrid-live-dev.private.wso2.com/admin/job/team-ei/job/dev_mi_docker/). 
The approach, 
- Taking the existing docker image as a base image which is already pushed to docker hub through MI Jenkins job. 
- Created a separate Dockerfile to pull the base image from the docker hub and copy the necessary testing artifacts to the relevant location inside the docker container. 
- Add the test.sh file to clone the artifacts(test cases) from the MI repo and run the tests. 

## User stories
https://github.com/wso2/micro-integrator/issues/169

## Automation tests
Tests wrote using Rest assured framework. 

## Samples
https://docs.wso2.com/display/EI650/Quick+Start+with+WSO2+Micro+Integrator

## Test environment
Docker Alpine

### CODE REVIEW

File: distribution/src/resources/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,8 @@
 
 # Base image
 
-FROM wso2/micro-integrator:latest
-
+FROM wso2/micro-integrato:1.1.0-SNAPSHOT
 COPY files/carbonapps /home/wso2carbon/wso2mi/repository/deployment/server/carbonapps
 
+
```
Comment (Nirothipan) [inline distribution/src/resources/dockerfiles/Dockerfile]: Shall we remove these empty lines

File: mi_docker_tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Nirothipan) [inline mi_docker_tests/pom.xml]: Shall we add the license header for this.

File: mi_docker_tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    <artifactId>mi_docker_tests</artifactId>
+
+    <packaging>jar</packaging>
+    <dependencies>
+        <!-- https://mvnrepository.com/artifact/io.rest-assured/rest-assured -->
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <version>3.1.0</version>
```
Comment (Nirothipan) [inline mi_docker_tests/pom.xml]: Shall we inherit the version from parent pom

File: mi_docker_tests/pom.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            <scope>test</scope>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/org.testng/testng -->
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>6.14.3</version>
```
Comment (Nirothipan) [inline mi_docker_tests/pom.xml]: Shall we inherit the version from parent pom

File: mi_docker_tests/pom.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+
```
Comment (Nirothipan) [inline mi_docker_tests/pom.xml]: Please remove these empty lines

File: mi_docker_tests/src/test/java/SampleTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    /**
+     * Tests to verify hello world sample.
+     */
+
+    @Test
+    public void sample_test_1() {
+        Response response=
+                given().
+                        when().
```
Comment (Nirothipan) [inline mi_docker_tests/src/test/java/SampleTest.java]: Shall we bring this to single line and format

File: mi_docker_tests/src/test/java/SampleTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ *    under the License.
+ */
+
+import io.restassured.response.Response;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static io.restassured.RestAssured.given;
+
+public class SampleTest {
```
Comment (Nirothipan) [inline mi_docker_tests/src/test/java/SampleTest.java]: Shall we add some meaningful class name and comments

File: mi_docker_tests/src/test/java/SampleTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+     * Tests to verify hello world sample.
+     */
+
+    @Test
+    public void sample_test_1() {
+        Response response=
+                given().
+                        when().
+                        get("http://localhost:8290/hello-world");
+        Assert.assertTrue(response.statusCode()==200);
```
Comment (Nirothipan) [inline mi_docker_tests/src/test/java/SampleTest.java]: Let's add a message to this.

File: mi_docker_tests/src/test/java/SampleTest.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        System.out.println("Response: "+response.getBody().asString());
```
Comment (Nirothipan) [inline mi_docker_tests/src/test/java/SampleTest.java]: Shall we remove this sys out

File: mi_docker_tests/src/test/resources/testng.xml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
```
Comment (Nirothipan) [inline mi_docker_tests/src/test/resources/testng.xml]: let's add the license header for this.

File: pom.xml [hunk 11]
```diff
@@ -47,6 +47,7 @@
         <module>p2-profile</module>
         <module>distribution</module>
         <module>integration</module>
+        <module>mi_docker_tests</module>
```
Comment (Nirothipan) [inline pom.xml]: Are we going to run this in normal profile , what if the docker is not installed.

**Comment by sdkottegoda** [comment]
Closing due to the conflicts. Please resolve conflicts and re-open the PR.

**Pull request closed without merging**